File: smartrecyclerview/src/main/java/com/lapptelier/smartrecyclerview/OnMoreListener.java
package com.lapptelier.smartrecyclerview;

/**
 * com.lapptelier.smartrecyclerview
 *
 * @author L'Apptelier SARL
 * @date 17/09/2017
 */
public interface OnMoreListener {
    /**
     * @param overallItemsCount
     * @param itemsBeforeMore
     * @param maxLastVisiblePosition for staggered grid this is max of all spans
     */
    void onMoreAsked(int overallItemsCount, int itemsBeforeMore, int maxLastVisiblePosition);
}

File: smartrecyclerview/src/main/java/com/lapptelier/smartrecyclerview/OnMoreListener.kt
package com.lapptelier.smartrecyclerview

/**
 * com.lapptelier.smartrecyclerview
 *
 * @author L'Apptelier SARL
 * @date 17/09/2017
 */
interface OnMoreListener {
    /**
     * @param overallItemsCount
     * @param itemsBeforeMore
     * @param maxLastVisiblePosition for staggered grid this is max of all spans
     */
    fun onMoreAsked(overallItemsCount: Int, itemsBeforeMore: Int, maxLastVisiblePosition: Int)
}
